src/main/java/com/revature/controllers/TripController.java

```diff
@@ -1,137 +1,219 @@
-
-        if (existingTrip.isPresent()) return ResponseEntity.status(201).body(tripService.updateTrip(trip));
-
-        return ResponseEntity.badRequest().build();
-    }
-
-    /**
-     * HTTP DELETE method (/trips/{id})
-     *
-     * @param id represents the Trip's ID.
-     * @return A string that says which Trip was deleted.
-     */
-    @ApiOperation(value = "Deletes a trip by ID", tags = {"Trip"})
-    @DeleteMapping("/{id}")
-    public ResponseEntity<String> deleteTripById(@PathVariable("id") int id) {
-        Optional<Trip> trip = tripService.getTripById(id);
-
-        if (trip.isPresent()) {
-            String message = tripService.deleteTripById(id);
-
-            return ResponseEntity.status(HttpStatus.ACCEPTED).body(message);
-        }
-
-        return ResponseEntity.notFound().build();
-    }
+	@Autowired
+	private TripService tripService;
+	@Autowired
+	private UserService userService;
+
+	/**
+	 * HTTP GET method (/trips)
+	 *
+	 * @return A list of all the trips.
+	 */
+	@ApiOperation(value = "Return all trips", tags = {"Trip"})
+	@GetMapping
+	public ResponseEntity<List<TripDTO>> getTrips() {
+		List<Trip> tripsRaw = tripService.getTrips();
+		List<TripDTO> trips = new ArrayList<>();
+		for (Trip trip : tripsRaw) {
+			trips.add(new TripDTO(trip));
+		}
+
+		if (!trips.isEmpty()) return ResponseEntity.ok(trips);
+
+		return ResponseEntity.noContent().build();
+	}
+
+	/**
+	 * HTTP GET method (/trips/{number})
+	 *
+	 * @param id represent the trip's ID.
+	 * @return A trip that matches the ID.
+	 */
+	@ApiOperation(value = "Return trip by ID", tags = {"Trip"})
+	@GetMapping("/{id}")
+	public ResponseEntity<TripDTO> getTripById(@PathVariable("id") int id) {
+		Optional<Trip> trip = tripService.getTripById(id);
+
+		if (trip.isPresent()) return ResponseEntity.ok(new TripDTO(trip.get()));
+
+		return ResponseEntity.noContent().build();
+	}
+
+	/**
+	 * HTTP GET method (/trips/driver/{driveId})
+	 *
+	 * @param driverId represents the driver's ID.
+	 * @return A list of trips matching the driver's ID.
+	 */
+	@ApiOperation(value = "Returns trips by driver ID", tags = {"Driver", "Trip"})
+	@GetMapping("/driver/{driverId}")
+	public ResponseEntity<List<TripDTO>> getTripsByDriverId(@PathVariable("driverId") int driverId) {
+		List<Trip> tripsRaw = tripService.getTripsByDriverId(driverId);
+		List<TripDTO> trips = new ArrayList<>();
+		for (Trip trip : tripsRaw) {
+			trips.add(new TripDTO(trip));
+		}
+
+		if (!trips.isEmpty()) return ResponseEntity.ok(trips);
+
+		return ResponseEntity.noContent().build();
+	}
+
+	/**
+	 * HTTP GET method (/trips/rider/{riderId})
+	 *
+	 * @param riderId represents the rider's ID.
+	 * @return A list of trips matching the rider's ID.
+	 */
+	@ApiOperation(value = "Returns trips by rider ID", tags = {"Rider", "Trip"})
+	@GetMapping("/rider/{riderId}")
+	public ResponseEntity<List<TripDTO>> getTripsByRiderId(@PathVariable("riderId") int riderId) {
```

Comment:
This is an unrelated change, but can we update this method to also utilize @ QueryParam instead of @ PathVariable. Path variables are great, but I'm of the opinion that it is RESTful when the path variable is used to identify the type of resource we are returning. In this case however, the path variable is used to identify the rider, which is only being used to further filter the Trips. This context makes more sense with an @ QueryParam.

Reply:
Working on this, it is going to turn into a major refactor, but Giblert and I agree that it will look much cleaner once changed.